Source/WebCore/loader/archive/cf/LegacyWebArchive.h

```diff
@@ -43,16 +43,30 @@ struct SimpleRange;
 
 class LegacyWebArchive final : public Archive {
 public:
+    // Archive is created directly from data or members so ArchiveOptions is not needed.
     WEBCORE_EXPORT static Ref<LegacyWebArchive> create();
-    WEBCORE_EXPORT static Ref<LegacyWebArchive> create(Ref<ArchiveResource>&& mainResource, Vector<Ref<ArchiveResource>>&& subresources, Vector<FrameIdentifier>&& subframeIdentifiers);
     WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(FragmentedSharedBuffer&);
     WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(const URL&, FragmentedSharedBuffer&);
+    WEBCORE_EXPORT static Ref<LegacyWebArchive> create(Ref<ArchiveResource>&& mainResource, Vector<Ref<ArchiveResource>>&& subresources, Vector<FrameIdentifier>&& subframeIdentifiers);
     WEBCORE_EXPORT static Ref<LegacyWebArchive> create(Ref<ArchiveResource>&& mainResource, Vector<Ref<ArchiveResource>>&& subresources, Vector<Ref<LegacyWebArchive>>&& subframeArchives);
+
     enum class ShouldArchiveSubframes : bool { No, Yes };
-    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(Node&, NOESCAPE const Function<bool(LocalFrame&)>& frameFilter = { }, const Vector<MarkupExclusionRule>& markupExclusionRules = { }, const String& mainFrameFileName = { }, bool saveScriptsFromMemoryCache = true, ShouldArchiveSubframes = ShouldArchiveSubframes::Yes);
-    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(LocalFrame&, ShouldArchiveSubframes = ShouldArchiveSubframes::Yes);
-    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> createFromSelection(LocalFrame*, bool saveScriptsFromMemoryCache = true, ShouldArchiveSubframes = ShouldArchiveSubframes::Yes);
-    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(const SimpleRange&, bool saveScriptsFromMemoryCache = true, ShouldArchiveSubframes = ShouldArchiveSubframes::Yes);
+    enum class ShouldSaveScriptsFromMemoryCache : bool { No, Yes };
+    struct ArchiveOptions {
+        ArchiveOptions()
+            : shouldArchiveSubframes(ShouldArchiveSubframes::Yes)
+            , shouldSaveScriptsFromMemoryCache(ShouldSaveScriptsFromMemoryCache::Yes)
+        {
+        }
+        ShouldArchiveSubframes shouldArchiveSubframes;
```

Comment:
We may be able to use a bool here. Now that this is no longer a parameter, we may not need to require an enum class here.